pkg/scheduler/framework/plugins/interpodaffinity/filtering_test.go

```diff
@@ -38,6 +37,15 @@ var (
 	defaultNamespace = ""
 )
 
+var cmpOpts = []cmp.Option{
```

Comment:
-1

`framework.Status` already implements `Equal`. We shouldn't re-implement it again.

Reply:
In general, `Equal` already does what you are trying to do.
The only difference is when it comes to errors. Here we use `errors.Is`. As long as you are comparing against pointers to specific errors or using an error type that implements `Is`, you don't need to compare strings.

Reply:
Yes, exactly as you said, but some statuses like `framework.AsStatus(errors.New("PVC test/abc-xyz was not created for pod test/abc (pod is not owner)"))`, it has a plain error then we have to define the error first or we'll fail in the `cmp.Diff()` , however, it's tedious and not that ideal, we have to define a lot of specific errors. 

Since `Equal` is only for testing, can we compare the error.String() directly?

Reply:
> `framework.Status` already implements `Equal`. We shouldn't re-implement it again.

I hadn't noticed before that `framework.Status` already had method `Equal`  to compare. It is right not to re-implement it again.

Some information：
>  using an error type that implements Is

Not available in test files in the current directory.
- `framework.Status.Equal` method has calling `errors.Is`  for comparison using the `cmpopts.EquateErrors()` argument. Error types `*fmt.warpError` simulated using `framework.AsStatus(fmt.Errorf())` in these tests do not implement the `errors.Is` method, so the return value is false
```
	if !cmp.Equal(s.err, x.err, cmpopts.EquateErrors()) {
		return false
	}
```

> As long as you are comparing against pointers to specific errors 

I don't understand your specific meaning. error is not exported from the `framework.Status` struct. Do you mean to compare error in these tests? similar to the following example：
```
	if diff := cmp.Diff(wantStatus, gotStatus, cmp.Comparer(func(a, b *framework.Status) bool {
		return reflect.DeepEqual(a, b)
	})); diff != "" {
		t.Errorf("Unexpected status (-want, +got):\n%s", diff)
	}
```
 or modify the `framework.Status.Equal` method?  Or does it mean something else? @alculquicondor

Reply:
> Since Equal is only for testing, can we compare the error.String() directly?

That is exactly what I'm trying to avoid. Comparing strings means that every time we improve an error message, we need to change unit tests, even if the functionality is the same. Yes, it might be tedious to migrate, but we only have to do it once.

> I don't understand your specific meaning. error is not exported from the framework.Status struct.

Yes, I mean that we can change that and export the necessary error.

> or modify the framework.Status.Equal method to compare err.Error()?

No, that compares strings.